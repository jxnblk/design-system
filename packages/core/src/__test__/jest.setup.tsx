import React from 'react'
import { ThemeProvider } from 'styled-components'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

import { createTheme } from '..'

global.rendererCreateWithTheme = (tree) =>
  renderer.create(<ThemeProvider theme={createTheme()}>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
