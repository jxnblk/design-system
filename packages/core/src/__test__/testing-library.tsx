import { render, RenderResult } from '@testing-library/react'
import PropTypes from 'prop-types'
import React from 'react'

import { ThemeProvider } from '..'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }

export const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms))
