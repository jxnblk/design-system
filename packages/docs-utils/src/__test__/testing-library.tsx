import { render, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'
import PropTypes from 'prop-types'
import React from 'react'

const WithThemeProvider = ({ children }) => <ThemeProvider>{children}</ThemeProvider>
WithThemeProvider.propTypes = {
  children: PropTypes.node,
}

// @ts-ignore
const customRender: (Element, any?) => RenderResult = (ui, options) =>
  render(ui, { wrapper: WithThemeProvider, ...options })

// re-export everything
export * from '@testing-library/react'
// override render method
export { customRender as render }
export const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
