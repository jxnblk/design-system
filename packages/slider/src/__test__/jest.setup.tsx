import React from 'react'
import { ThemeProvider } from 'pcln-design-system'

import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'
import renderer from 'react-test-renderer'

global.rendererCreateWithTheme = (tree) => renderer.create(<ThemeProvider>{tree}</ThemeProvider>)

global.requestAnimationFrame =
  global.requestAnimationFrame ||
  function _raf(cb) {
    return setTimeout(cb, 0)
  }
