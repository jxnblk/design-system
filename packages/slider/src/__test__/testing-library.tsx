import type { RenderOptions, RenderResult } from '@testing-library/react'
import type { ReactElement } from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
