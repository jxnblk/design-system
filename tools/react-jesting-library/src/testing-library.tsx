import React, { FC, ReactElement } from 'react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import { ThemeProvider } from 'pcln-design-system'

const AllTheProviders: FC = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>
}

/**
 * customRender wraps the render method of \@testing-library/react and wraps it in our
 * custom AllTheProviders which provides a pcln-design-system theme
 *
 * @public
 */
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
