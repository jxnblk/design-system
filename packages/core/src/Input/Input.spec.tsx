import React from 'react'
import { fireEvent, render, screen } from '../__test__/testing-library'
import ForwardRefDemo from '../storybook/utils/ForwardRefsDemo'

import { Button, Input, theme, createTheme, getPaletteColor } from '..'

const id = 'fake-test-id'

describe('Input', () => {
  test('it renders', () => {
    const json = rendererCreateWithTheme(<Input id={id} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a red border with a color prop is passed', () => {
    const json = rendererCreateWithTheme(<Input id={id} color='red' />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with a really large padding and margin', () => {
    const json = rendererCreateWithTheme(<Input id={id} p={4} m={4} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('it renders an input element with large text', () => {
    const json = rendererCreateWithTheme(<Input id={id} fontSize={4} />).toJSON()
    expect(json).toMatchSnapshot()
  })
  test('forwards ref to the input', () => {
    const target = 'TARGET'
    const placeholder = 'Apples'
    function refChild(dsRef) {
      function onClick() {
        dsRef.current.focus()
      }
      return (
        <>
          <Input id={id} ref={dsRef} placeholder={placeholder} />
          <Button onClick={onClick} mt={3}>
            {target}
          </Button>
        </>
      )
    }
    const { getByText, getByPlaceholderText } = render(<ForwardRefDemo refChild={refChild} />)
    fireEvent.click(getByText(target))
    expect(getByPlaceholderText(placeholder)).toHaveFocus()
  })

  test('it renders disabled', () => {
    const paletteTheme = createTheme(theme)
    render(<Input id={id} disabled placeholder='Disabled Input' />)

    const input = screen.getByPlaceholderText('Disabled Input')
    expect(input).toBeDisabled()
    expect(input).toHaveStyleRule(
      'background-color',
      getPaletteColor('background.light')({ theme: paletteTheme }),
      { modifier: ':disabled' }
    )
    expect(input).toHaveStyleRule('color', getPaletteColor('text.light')({ theme: paletteTheme }), {
      modifier: ':disabled',
    })
    expect(input).toHaveStyleRule('cursor', 'not-allowed', { modifier: ':disabled' })
  })

  describe('helper text', function () {
    test('renders with same color as Input', () => {
      const helperText = 'hello world'
      const color = 'error.base'
      const paletteTheme = createTheme(theme)
      const { getByText } = render(
        <Input id={id} color={color} helperText={<Input.HelperText>{helperText}</Input.HelperText>} />
      )
      const helperTextNode = getByText(helperText)
      expect(helperTextNode).not.toBeNull()
      expect(helperTextNode).toHaveStyleRule('color', getPaletteColor(color)({ theme: paletteTheme }))
    })

    test('can override the color from Input', () => {
      const helperText = 'hello world'
      const placeholder = 'click me'
      const color = 'error.base'
      const helperTextColor = 'secondary.base'
      const paletteTheme = createTheme(theme)
      const { getByText, getByPlaceholderText } = render(
        <Input
          id={id}
          color={color}
          placeholder={placeholder}
          helperText={<Input.HelperText color={helperTextColor}>{helperText}</Input.HelperText>}
        />
      )
      const helperTextNode = getByText(helperText)
      expect(helperTextNode).not.toBeNull()
      expect(helperTextNode).toHaveStyleRule(
        'color',
        getPaletteColor(helperTextColor)({ theme: paletteTheme })
      )
      expect(getByPlaceholderText(placeholder)).toHaveStyleRule(
        'border-color',
        getPaletteColor(color)({ theme: paletteTheme })
      )
    })
  })
})
