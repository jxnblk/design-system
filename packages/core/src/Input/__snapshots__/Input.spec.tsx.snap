// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input it renders 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 16px;
  border-radius: 12px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:disabled,
.c1:-moz-read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:disabled,
.c1:read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 14px;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    fontSize={
      [
        2,
        null,
        1,
      ]
    }
    id="fake-test-id"
    size="lg"
  />
</div>
`;

exports[`Input it renders an input element with a really large padding and margin 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  padding: 32px;
  margin: 32px;
  font-size: 16px;
  border-radius: 12px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:disabled,
.c1:-moz-read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:disabled,
.c1:read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 14px;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    fontSize={
      [
        2,
        null,
        1,
      ]
    }
    id="fake-test-id"
    size="lg"
  />
</div>
`;

exports[`Input it renders an input element with a red border with a color prop is passed 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c00;
  font-size: 16px;
  border-radius: 12px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:disabled,
.c1:-moz-read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:disabled,
.c1:read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:focus {
  outline: 0;
  border-color: #c00;
  box-shadow: 0 0 0 2px #c00;
}

@media screen and (min-width:32em) {

}

@media screen and (min-width:40em) {
  .c1 {
    font-size: 14px;
  }
}

<div
  className="c0"
>
  <input
    className="c1"
    color="red"
    fontSize={
      [
        2,
        null,
        1,
      ]
    }
    id="fake-test-id"
    size="lg"
  />
</div>
`;

exports[`Input it renders an input element with large text 1`] = `
.c0 {
  font-family: 'Montserrat','Helvetica Neue',Helvetica,Arial,sans-serif;
  line-height: 1.4;
  font-weight: 500;
}

.c0 * {
  box-sizing: border-box;
}

.c1 {
  position: relative;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  color: #001833;
  background-color: transparent;
  border-width: 1px;
  border-style: solid;
  margin: 0;
  padding: 14px 12px 14px 12px;
  border-color: #c0cad5;
  font-size: 24px;
  border-radius: 12px;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:disabled,
.c1:-moz-read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:disabled,
.c1:read-only {
  background-color: #f4f6f8;
  color: #4f6f8f;
  cursor: not-allowed;
}

.c1:focus {
  outline: 0;
  border-color: #0068ef;
  box-shadow: 0 0 0 2px #0068ef;
}

<div
  className="c0"
>
  <input
    className="c1"
    fontSize={4}
    id="fake-test-id"
    size="lg"
  />
</div>
`;
